Modal.js

```diff
@@ -0,0 +1,155 @@
+                
+        const modalTitle = parsedContent.querySelector('.modal-card-title')
+        modalTitle.innerText = this.title
+
+        const modalMessage = parsedContent.querySelector('.label')
+        modalMessage.innerText = this.message
+
+        if (this.requireUserInput) {
+
+            const parsedInput = parser.parseFromString(this.input, 'text/html')
+            const input = parsedInput.querySelector('input')
+            const parsedSection = parsedContent.getElementById('modal-card-field')
+            parsedSection.append(input)
+
+        }
+
+        const modalPrimaryButton = parsedContent.getElementById('modal-primary-button')
+        modalPrimaryButton.innerText = this.primaryBtnText
+
+        const modalSecondaryButton = parsedContent.getElementById('modal-secondary-button')
+        modalSecondaryButton.innerText = this.secondaryBtnText
+
+        const modalDiv = parsedContent.querySelector('.modal')
+        document.body.appendChild(modalDiv)
+
+        this.#createCloseEvents() // Listens for events to close the modal
+
+    }
+
+    prompt(funcToRunOnOkay, funcToRunOnCancel) {
+
+        this.requireUserInput = true
+        this.#createTemplate()
+
+        const primaryButton = document.getElementById('modal-primary-button')
+        const secondaryButton = document.getElementById('modal-secondary-button')
+        const userInput = document.getElementById('modal-input')
+
+        if (funcToRunOnOkay) {
+        
+            primaryButton.addEventListener('click', () => {
+
+                funcToRunOnOkay(userInput.value)
+            
+            })
+
+        }
+
+        secondaryButton.addEventListener('click', () => {
+
+            if (funcToRunOnCancel) {
+
+                funcToRunOnCancel(userInput.value)
+                
+            } else { this.#close() }
+
+        })
+
+    }
+
+    confirm(funcToRunOnOkay, funcToRunOnCancel) {
+
+        this.#createTemplate()
+
+        const primaryButton = document.getElementById('modal-primary-button')
+        const secondaryButton = document.getElementById('modal-secondary-button')
+
+        if (funcToRunOnOkay) {
+
+            primaryButton.addEventListener('click', () => {
+
+                funcToRunOnOkay(true)
+
+            })
+
+        }
+
+        secondaryButton.addEventListener('click', () => {
+
+            if (funcToRunOnCancel) {
+
+                funcToRunOnCancel(userInput.value)
+                
+            } else { this.#close() }
+
+        })
+
+    }
+
+    #createCloseEvents() {
+
+        document.querySelectorAll('.modal-background, .delete, .modal-close').forEach((el) => {
+
+            el.addEventListener('click', () => { this.#close() })
+
+        }) 
+
+        document.addEventListener('keydown', (event) => { event.key === "Escape" && this.#close()})
+
+    }
+
+    #close() {
+
+        const modal = document.getElementById(this.#id)
+
+        modal && modal.remove()
```

Comment:
Do you think the one liner `if (modal) modal.remove()` is more readable than using a logical operator combined with short circuiting logical evaluation?

Reply:
I'm afraid I can't respond since I have died on this hill.

Reply:
😂😂